homeassistant/components/fmi/__init__.py

```diff
@@ -0,0 +1,138 @@
+"""The FMI (Finnish Meteorological Institute) component."""
+
+import asyncio
+from datetime import date, datetime
+
+from async_timeout import timeout
+from dateutil import tz
+import fmi_weather_client as fmi
+from fmi_weather_client.errors import ClientError, ServerError
+
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_OFFSET,
+    SUN_EVENT_SUNSET,
+)
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.sun import get_astral_event_date
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    _LOGGER,
+    COORDINATOR,
+    DOMAIN,
+    FMI_WEATHER_SYMBOL_MAP,
+    MIN_TIME_BETWEEN_UPDATES,
+    UNDO_UPDATE_LISTENER,
+)
+
+PLATFORMS = ["weather"]
+
+
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
+    """Set up configured FMI."""
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
in the latest version, this line moves the async_setup_entry and the async_setup method can be deleted.